app/jobs/socure_docv_results_job.rb

```diff
@@ -3,35 +3,50 @@
 class SocureDocvResultsJob < ApplicationJob
   queue_as :default
 
-  attr_reader :document_capture_session_uuid
+  attr_reader :document_capture_session_uuid, :async
 
   # @param [String] document_capture_session_uuid
-  def perform(document_capture_session_uuid:)
+  def perform(document_capture_session_uuid:, async: true)
     @document_capture_session_uuid = document_capture_session_uuid
+    @async = async
 
-    dcs = DocumentCaptureSession.find_by(uuid: document_capture_session_uuid)
-    raise "DocumentCaptureSession not found: #{document_capture_session_uuid}" if !dcs
+    raise "DocumentCaptureSession not found: #{document_capture_session_uuid}" unless
+      document_capture_session
 
-    # analytics = create_analytics(
-    #   user: dcs.user,
-    #   service_provider_issuer: dcs.issuer,
-    # )
-
-    result = socure_document_verification_result
-    dcs.store_result_from_response(result)
+    timer = JobHelpers::Timer.new
+    response = timer.time('vendor_request') do
+      socure_document_verification_result
+    end
+    log_verification_request(
+      docv_result_response: response,
+      vendor_request_time_in_ms: timer.results['vendor_request'],
+    )
+    document_capture_session.store_result_from_response(response)
   end
 
   private
 
-  def create_analytics(
-    user:,
-    service_provider_issuer:
-  )
-    Analytics.new(
-      user:,
+  def analytics
+    @analytics ||= Analytics.new(
+      user: document_capture_session.user,
       request: nil,
-      sp: service_provider_issuer,
       session: {},
+      sp: document_capture_session.issuer,
+    )
+  end
+
+  def log_verification_request(docv_result_response:, vendor_request_time_in_ms:)
+    return if docv_result_response.nil? || document_capture_session.nil?
```

Comment:
```suggestion
```
* we check that document capture session exists in the construction
* if we make the call then we need to log it ...even it the response is nil or throws an exception because a req'd field is not present